packages/examples/cvat/recording-oracle/src/handlers/process_intermediate_results.py

```diff
@@ -980,17 +405,126 @@ def process_intermediate_results(  # noqa: PLR0912
             ", ".join(f"{k}: {v:.2f}" for k, v in job_results.items()),
         )
 
-    if validation_result.updated_gt_stats:
-        updated_gt_stats = _compute_gt_stats_update(
-            initial_gt_stats, validation_result.updated_gt_stats
-        )
+    gt_stats = validation_result.gt_stats
+    if gt_stats:
+        # cvat_task_id: {val_frame_id, ...}
+        cvat_task_id_to_failed_val_frames: dict[int, set[int]] = {}
+        rejected_job_ids = rejected_jobs.keys()
+
+        if rejected_job_ids:
+            job_id_to_task_id = {j.job_id: j.task_id for j in unchecked_jobs_meta.jobs}
+            job_id_to_frame_range = {j.job_id: j.job_frame_range for j in unchecked_jobs_meta.jobs}
+
+            # find validation frames to be disabled
+            for rejected_job_id in rejected_job_ids:
+                job_frame_range = job_id_to_frame_range[rejected_job_id]
+                cvat_task_id = job_id_to_task_id[rejected_job_id]
+                honeypots_mapping = validation_result.task_id_to_honeypots_mapping[cvat_task_id]
+                job_honeypots = set(honeypots_mapping.keys()) & set(job_frame_range)
+                validation_frames = [
+                    val_frame
+                    for honeypot, val_frame in honeypots_mapping.items()
+                    if honeypot in job_honeypots
+                ]
+                sorted_task_frame_names = validation_result.task_id_to_sequence_of_frame_names[
+                    cvat_task_id
+                ]
+
+                for val_frame in validation_frames:
+                    val_frame_name = sorted_task_frame_names[val_frame]
+                    val_frame_stats = gt_stats[val_frame_name]
+                    if (
+                        val_frame_stats.failed_attempts >= Config.validation.gt_ban_threshold
+                        and not val_frame_stats.accepted_attempts
+                    ):
+                        cvat_task_id_to_failed_val_frames.setdefault(cvat_task_id, set()).add(
+                            val_frame
+                        )
+
+        for cvat_task_id, val_frame_ids in cvat_task_id_to_failed_val_frames.items():
+            task_validation_layout = validation_result.task_id_to_val_layout[cvat_task_id]
+            intersection = val_frame_ids & set(task_validation_layout.disabled_frames)
+
+            if intersection:
+                logger.error(
+                    "Logical error occurred while disabling validation frames "
+                    f"for the task({task_id}). Frames {intersection} "
+                    "are already disabled."
+                )
+
+            updated_disable_frames = task_validation_layout.disabled_frames + list(val_frame_ids)
+            non_disabled_frames = list(
+                set(task_validation_layout.validation_frames) - set(updated_disable_frames)
+            )
+
```

Comment:
It makes sense to check there that there are at least `frames_per_job_count` items in `non_disabled_frames`. In this case the task (and, probably, the escrow) becomes unverifiable.

Reply:
It's checked when we try to select new N validation frames from the `non_disabled_frames`. In that case, we do not shuffle honeypots (updated_task_honeypot_real_frames is not changed)